phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyWithMaxLookbackIT.java

```diff
@@ -0,0 +1,206 @@
+            conn.commit();
+            long afterInsertSCN = EnvironmentEdgeManager.currentTimeMillis() + 1;
+            testClock.setValue(afterInsertSCN);
+            EnvironmentEdgeManager.injectEdge(testClock);
+            testClock.incrementValue(1);
+            testClock.incrementValue(MAX_LOOKBACK /2  * 1000);
+            long scrutinyTs = EnvironmentEdgeManager.currentTimeMillis();
+            changeIndexValue();
+            testClock.incrementValue(MAX_LOOKBACK /2  * 1000);
+            List<Job> completedJobs = runScrutiny(schema, tableName, indexTableName, scrutinyTs);
+            Job job = completedJobs.get(0);
+            assertTrue(job.isSuccessful());
+            assertCounters(job.getCounters());
+        }
+    }
+    private void changeIndexValue() throws SQLException {
+        try (Connection conn =
+                 DriverManager.getConnection(getUrl(), PropertiesUtil.deepCopy(TEST_PROPERTIES))) {
+            String tableName = isViewIndex ?
+                SchemaUtil.getTableName(schema, viewName) : dataTableFullName;
+            PreparedStatement stmt = getUpsertDataStmt(tableName, conn);
+            NonParameterizedIndexScrutinyToolIT.upsertRow(stmt, 1, "name-1", 38139);
+            conn.commit();
+        }
+    }
+
+    private static PreparedStatement getUpsertDataStmt(String tableFullName, Connection conn) throws SQLException {
+        return conn.prepareStatement(String.format(UPSERT_DATA, tableFullName));
+    }
+
+    private void assertCounters(Counters counters) {
+        assertEquals(1, getCounterValue(counters, VALID_ROW_COUNT));
+        assertEquals(1, getCounterValue(counters, BEYOND_MAX_LOOKBACK));
+        assertEquals(0, getCounterValue(counters, INVALID_ROW_COUNT));
+    }
+
+    private List<Job> runScrutiny(String schemaName, String dataTableName, String indexTableName,
+                                  Long scrutinyTs)
+        throws Exception {
+        return runScrutiny(schemaName, dataTableName, indexTableName, null, null, scrutinyTs);
+    }
+
+    private List<Job> runScrutiny(String schemaName, String dataTableName, String indexTableName,
+                                  Long batchSize, IndexScrutinyTool.SourceTable sourceTable,
+                                  Long scrutinyTs) throws Exception {
+        final String[]
+            cmdArgs =
+            getArgValues(schemaName, dataTableName, indexTableName, batchSize, sourceTable,
+                false, null, null, null, scrutinyTs);
+        return runScrutiny(MaxLookbackIndexScrutinyMapper.class, cmdArgs);
+    }
+
+    private static class MaxLookbackIndexScrutinyMapper extends IndexScrutinyMapper {
+        @Override
+        public void postSetup(){
+            try {
+                String tableToCompact;
+                if (isViewIndex){
+                    String physicalDataTableName =
+                        SchemaUtil.getPhysicalHBaseTableName(schema, dataTableName, false).getString();
```

Comment:
why not take isNamespaceEnabled property from the config?

Reply:
@gjacoby126

Reply:
I know it's going to be false.